### ISSUE TITLE
Issues with Rojas boundary file

### ISSUE THREAD

**Issue by emilykbedwell** [leombastos/2022_AdvPA#2]
I am working with the boundary file (I have converted it from a shapefile to a geojson file) and the plots don't look right. It is only showing one longitude metric on the y axis, so it is making the field look thin and distorted. Also, the only data that shows on the "sf" table is 2 rows and 2 columns. the columns are named x_lflt_, layerId, and geometry. The only data in these columns is the number 89 in the x_lflt_ column. Here are some pics. I feel like there should be more data here--  Has anyone dealt with a similar issue, yet?  Thanks!! 
<img width="694" alt="github issue1b" src="https://user-images.githubusercontent.com/97464584/166507483-e4cc8e64-9fe1-4ce0-87b6-0630791c8896.PNG">
<img width="720" alt="github issue1" src="https://user-images.githubusercontent.com/97464584/166507485-9c2fd993-9546-4a37-a9f8-9160eb4386bf.PNG">

**Comment by leombastos** [comment]
Hi Emily, thanks for your question.
Would you be able to provide a bit more info in what was the issue, and how you fixed it?

Your thoughts can be useful for someone that may run into a similar issue.

Thanks, and nice job troubleshooting it!

Leo

**Comment by leombastos** [comment]
Nice! Thanks Emily for sharing the issue and the solution! And thanks Giannis for assisting!

**Comment by leombastos** [comment]
Although it has been fixed, I will leave the issue "opened" so it is easier for others to see if they run into something similar.

**Comment by spenoliver** [comment]
> The issue with the 2 rows is due to there being 2 separate polygons in the boundary file. I solved this issue (with Giannis' help :) ) by merging the two polygons into one.

Hey Emily! How did you go about merging the polygons? Because I am running into a similar issue

**Comment by spenoliver** [comment]
Wait I just did something with the st_union function and it appears to have worked. I don't know if that is what I was supposed to do

**Comment by leombastos** [comment]
Hi Spen.

When I was preparing the project, I did not need to merge the polygons to get it to work.

There is no single correct way of doing this, thus Emily's and Giannis' approach to merging them was a nice workaround. Nonetheless, if it worked for you without needing to merge the polygons, then that's ok too (as in my case, I didn't need to merge them to get the class code to work).

**Comment by emilykbedwell** [comment]
Hey, Spen! Sorry for the delay! As Dr. Bastos said, the polygons don't have to be merged for it to work (the read_sf works for shapefiles even when there are multiple ones-- as far as I know). Good job figuring it out!

**Comment by spenoliver** [comment]
The code would run fine, but when I went to filter with the buffer it doubled the data points so I had over 120000 points instead of reducing by the number that was outside the perimeter which I thought was odd

**Comment by emilykbedwell** [comment]
Spen, that is strange. Did you use a negative number (between 20 and maybe 40m) for the buffer? it should decrease the number of observations (data points). Can you show your code?

**Comment by Galio23** [comment]
Spen, keep in mind sometimes buffer won't really decrease your observations. It shouldn't increase them for sure.

**Comment by spenoliver** [comment]
Once I merged the boundaries with st_union the data was more reasonable
![Screen Shot 2022-05-06 at 10 34 04 AM](https://user-images.githubusercontent.com/105007869/167154891-a5aabd57-f915-426c-863b-aff01c697a2d.png)

**Comment by emilykbedwell** [comment]
You should be good to go with this!

**Comment by Galio23** [comment]
Move on with the buffer plot chunk and position filtering. Then check on the plot to see where the "yellowish" line of the buffer appears. This should give you an idea whether things are going well or not.

**Comment by tbell32** [comment]
Hi! Can anyone tell me how I merge the polygons?

Thanks

**Comment by tbell32** [comment]
Hey guys, can anyone tell me how to write the st_union function into the chunk?

**Comment by leombastos** [comment]
Hi Tison,

In class, we did not need to use `st_union()`. When I ran the project files on my side, I did not need to use it either. I was able to re-use the class code without issues.

What issue are you specifically getting? Can you share your code, and the error you are facing?

**Comment by tbell32** [comment]
I am getting an error when I run the filtered summary chunk
![tempImageEfJA0o](https://user-images.githubusercontent.com/98991780/167319134-e634d479-fb1e-4cb8-9a3e-c818c7b2e427.gif)
.

**Comment by leombastos** [comment]
What do you get when you run `class(Deer16_w_pf)`?

Can you show me the output when you simply print `Deer16_w_pf`?

**Comment by tbell32** [comment]
![tempImage3esI11](https://user-images.githubusercontent.com/98991780/167319462-3e8e7053-5197-4b1a-a78c-7cbd6a0176ab.gif)
![tempImage6H86FK](https://user-images.githubusercontent.com/98991780/167319465-f366f318-9dba-4adf-9233-64a1d0a1cbb3.gif)

**Comment by leombastos** [comment]
Ok, so the issue you are getting is because the object `Deer16_w_pf` has zero rows. 

You would need to go back to each of the chunks that were used to create `Deer16_w_pf` and see where the issue of zero rows first appeared. It is possible that it appeared whenever you used the `filter()` function. 

When you find which chunk created a zero-row df, that is the one you would need to inspect more closely to understand why you are getting zero rows back.

I used the class code for the project and didn't run into this issue. It is possible that maybe you had an object name or column name mispellled at some point and that may be the cause.

**Comment by tbell32** [comment]
Okay, thank you so much

**Comment by tbell32** [comment]
Hey Dr. Bastos my code ran fine when I changed the geojason file that I converted from the boundary data and replaced it with the boundary shape file.

**Comment by leombastos** [comment]
Nice Tison!

Are things flowing right? Would you still like to connect?

**Comment by sarahgjohnson** [comment]
Hi all, 

My code is running completely fine with no errors but I also have a huge increase in observations (73k to 121k) after filtering so I'm not really sure how to handle it.

**Comment by leombastos** [comment]
@spenoliver @tbell32 cpuld you share some thoughts on how you handled this issue? @sarahgjohnson  could you share a screenshot of your code?

**Comment by sarahgjohnson** [comment]
I think my issue is the same as spen's but without the polygon issue. My code doesn't have errors, but this is what my results look like before and after filtering
=
<img width="946" alt="Screen Shot 2022-05-10 at 2 26 27 PM" src="https://user-images.githubusercontent.com/104635938/167709552-f7047606-e23c-4f99-ab0f-644eb7e21e0d.png">

**Comment by sarahgjohnson** [comment]
@leombastos

**Comment by spenoliver** [comment]
Yes mine was the same it ran fine except for the data doubled. What I did was bounary_w_merge <- st_union(boundary_w) after I uploaded the boundary object and it did not double to points when I went to filter. Tison did not merge the boundaries, I think his was able to run fine with the shp file when he uploaded it

**Comment by sarahgjohnson** [comment]
<img width="774" alt="Screen Shot 2022-05-10 at 3 50 41 PM" src="https://user-images.githubusercontent.com/104635938/167710615-49d978d8-9fde-456b-8c76-244fbaade391.png">
now it is giving this? @spenoliver

**Comment by sarahgjohnson** [comment]
no

**Comment by spenoliver** [comment]
![Screen Shot 2022-05-10 at 3 55 35 PM](https://user-images.githubusercontent.com/105007869/167711420-7e21b6eb-31e8-4818-be25-46045895bfc6.png)
This is what mine looks like

**Comment by spenoliver** [comment]
Thats odd because even if the union didn't work the new object should have been created if there were no errors

**Comment by sarahgjohnson** [comment]
<img width="768" alt="Screen Shot 2022-05-10 at 4 04 51 PM" src="https://user-images.githubusercontent.com/104635938/167712661-8a2e8b0d-e2ec-49b8-bfb2-71a2c01ed3c2.png">
yeah I'm not sure but it's totally geeking on me now.

**Comment by sarahgjohnson** [comment]
is it bc mine isnt a geojson file? I'm not sure how to convert it to that either

**Comment by spenoliver** [comment]
That could be it. https://mygeodata.cloud/converter/shp-to-geojson was the website I used to switch it. you have to make sure and upload all related files it needs to make the conversion. You may also want to keep an eye on what objects you reference because in that chunk you create just the boundary and not boundary_w

**Comment by spenoliver** [comment]
I will say I am crossing all my fingers hoping my finished product is reasonable, so I don't want to lead you astray because I am not 100% confident in what I did. Haha

**Comment by sarahgjohnson** [comment]
got it to work!! thanks so much spen! it went from 73k to 61k so hopefully that is better lol

**Comment by spenoliver** [comment]
Yeah that's about what my observations went to. No problem!

**Comment by sarahgjohnson** [comment]
@spenoliver when I uploaded the clean dataset into the interpolation code it is pulling back up as 121k rows... did that happen to you?

**Comment by spenoliver** [comment]
No I dont remember that happening. One thing that stinks is that you'll have to merge the geojson everytime you bring in the boundary file, so that might be it

**Comment by sarahgjohnson** [comment]
I figured it out! Had a weird save
